src/ray/raylet/node_manager.cc

```diff
@@ -650,12 +650,25 @@ void NodeManager::HandleActorStateTransition(const ActorID &actor_id,
     }
   } else if (actor_registration.GetState() == ActorTableData::DEAD) {
     // When an actor dies, loop over all of the queued tasks for that actor
-    // and treat them as failed.
+    // and treat them as failed
     auto tasks_to_remove = local_queues_.GetTaskIdsForActor(actor_id);
     auto removed_tasks = local_queues_.RemoveTasks(tasks_to_remove);
     for (auto const &task : removed_tasks) {
       TreatTaskAsFailed(task, ErrorType::ACTOR_DIED);
     }
+    // Loop over all actors and kill those whose parent actor
+    // is the actor that just died.
+    for (const auto &actor_entry : actor_registry_) {
+      auto node_manager_id = actor_entry.second.GetNodeManagerId();
+      // Kill this actor if it is local and is a child of the dead actor.
+      if (node_manager_id == gcs_client_->client_table().GetLocalClientId() && 
+          actor_entry.second.GetParentActorID() == actor_id) {
+        auto worker = worker_pool_.GetActorWorker(actor_entry.first);
```

Comment:
We should probably add a `RAY_CHECK(worker)` here with an appropriate fatal check message.